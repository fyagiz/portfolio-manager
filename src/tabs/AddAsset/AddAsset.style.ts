import { StyleSheet } from "react-native";
import { Colors, styleConstants } from "../../utils";

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.backgroundColor,
  },
  insideContainer: {
    backgroundColor: Colors.backgroundColor,
    borderRadius: styleConstants.borderRadius,
    borderWidth: styleConstants.borderRadius,
    padding: styleConstants.padding,
  },
  textInputContainer: {
    borderWidth: 2,
    padding: styleConstants.padding,
  },
  textInputStyle: {
    borderRadius: styleConstants.borderRadius,
    borderWidth: 2,
    padding: styleConstants.padding,
  },
});
