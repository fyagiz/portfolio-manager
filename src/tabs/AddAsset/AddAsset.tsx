import { Text, TextInput, View } from "react-native";
import styles from "./AddAsset.style";
import { useState } from "react";

const AddAsset = () => {
  const [stockName, setStockName] = useState<string>("");
  const [amount, setAmount] = useState<string>("");
  const [price, setPrice] = useState<string>("");
  const [commision, setCommision] = useState<string>("");

  const handleFormatInput = (text: string, setFunction: React.Dispatch<React.SetStateAction<string>>, isOnlyNumber: boolean = true) => {
    let filteredText: string;
    if (isOnlyNumber) {
      filteredText = text.replace(/[^0-9]/g, "");
    } else {
      filteredText = text.replace(/[^a-zA-Z]/g, "");
    }
    setFunction(filteredText);
  };
  return (
    <View style={styles.container}>
      <View style={styles.insideContainer}>
        <View style={styles.textInputContainer}>
          <Text testID="stockNameText">Stock Name</Text>
          <TextInput testID="stockNameTextInput" onChangeText={text => handleFormatInput(text, setStockName, false)} value={stockName} style={styles.textInputStyle} keyboardType="ascii-capable" />
          <Text testID="amountText">Amount</Text>
          <TextInput testID="amountTextInput" onChangeText={text => handleFormatInput(text, setAmount)} value={amount} style={styles.textInputStyle} keyboardType="number-pad" />
          <Text testID="priceText">Price</Text>
          <TextInput testID="priceTextInput" onChangeText={text => handleFormatInput(text, setPrice)} value={price} style={styles.textInputStyle} keyboardType="decimal-pad" />
          <Text testID="comissionText">Commission</Text>
          <TextInput testID="comissionTextInput" onChangeText={text => handleFormatInput(text, setCommision)} value={commision} style={styles.textInputStyle} keyboardType="decimal-pad" />
          <Text testID="comissionText">Commission</Text>
          <TextInput testID="comissionTextInput" onChangeText={text => handleFormatInput(text, setCommision)} value={commision} style={styles.textInputStyle} keyboardType="decimal-pad" />
        </View>
      </View>
    </View>
  );
};

export default AddAsset;
