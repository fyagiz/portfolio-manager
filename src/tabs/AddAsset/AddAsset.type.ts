export type HandleFormatInputFormatType = "stockCode" | "amount" | "price";
