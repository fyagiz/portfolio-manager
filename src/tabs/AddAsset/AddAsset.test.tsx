import { render } from "@testing-library/react-native";
import AddAsset from "./AddAsset";

/*
1- AddAsset should be rendered
*/

describe("AddAssetModal Unit Tests", () => {
  test("should render correctly", () => {
    const addAssetModal = render(<AddAsset />);
    expect(addAssetModal).toMatchSnapshot();
  });
});
