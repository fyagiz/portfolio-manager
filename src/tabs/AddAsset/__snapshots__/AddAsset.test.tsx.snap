// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAssetModal Unit Tests should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#017387",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#017387",
        "borderRadius": 4,
        "borderWidth": 4,
        "padding": 4,
      }
    }
  >
    <View
      style={
        {
          "borderWidth": 2,
          "padding": 4,
        }
      }
    >
      <Text
        testID="stockNameText"
      >
        Stock Name
      </Text>
      <TextInput
        keyboardType="ascii-capable"
        onChangeText={[Function]}
        style={
          {
            "borderRadius": 4,
            "borderWidth": 2,
            "padding": 4,
          }
        }
        testID="stockNameTextInput"
        value=""
      />
      <Text
        testID="amountText"
      >
        Amount
      </Text>
      <TextInput
        keyboardType="number-pad"
        onChangeText={[Function]}
        style={
          {
            "borderRadius": 4,
            "borderWidth": 2,
            "padding": 4,
          }
        }
        testID="amountTextInput"
        value=""
      />
      <Text
        testID="priceText"
      >
        Price
      </Text>
      <TextInput
        keyboardType="decimal-pad"
        onChangeText={[Function]}
        style={
          {
            "borderRadius": 4,
            "borderWidth": 2,
            "padding": 4,
          }
        }
        testID="priceTextInput"
        value=""
      />
      <Text
        testID="comissionText"
      >
        Commission
      </Text>
      <TextInput
        keyboardType="decimal-pad"
        onChangeText={[Function]}
        style={
          {
            "borderRadius": 4,
            "borderWidth": 2,
            "padding": 4,
          }
        }
        testID="comissionTextInput"
        value=""
      />
      <Text
        testID="comissionText"
      >
        Commission
      </Text>
      <TextInput
        keyboardType="decimal-pad"
        onChangeText={[Function]}
        style={
          {
            "borderRadius": 4,
            "borderWidth": 2,
            "padding": 4,
          }
        }
        testID="comissionTextInput"
        value=""
      />
    </View>
  </View>
</View>
`;
